src/crypto/crypto_kdf.cc

```diff
@@ -0,0 +1,131 @@
+#include "crypto/crypto_kdf.h"
+#include "ncrypto.h"
+#include "v8.h"
+
+#include <openssl/core_names.h>     /* OSSL_KDF_* */
+
+namespace node {
+namespace crypto {
+
+v8::MaybeLocal<v8::Value> KDFTraits::EncodeOutput(
+    Environment* env,
+    const KDFConfig& params,
+    ByteSource* out) {
+  return out->ToArrayBuffer(env);
+}
+
+v8::Maybe<void> KDFTraits::AdditionalConfig(
+    CryptoJobMode mode,
+    const v8::FunctionCallbackInfo<v8::Value>& args,
+    unsigned int offset,
+    KDFConfig* params) {
+  Environment* env = Environment::GetCurrent(args);
+
+  params->mode = mode;
+
+  CHECK(args[offset]->IsString());  // Algorithm
+  CHECK(args[offset + 1]->IsObject());  // Params
+  CHECK(args[offset + 2]->IsUint32());  // Length
+
+  Utf8Value algorithm(env->isolate(), args[offset]);
+  params->kdfCtx = ncrypto::KdfCtxPointer::FromName(algorithm.ToStringView());
+
+  // if (!pass.CheckSizeInt32()) [[unlikely]] {
+  //   THROW_ERR_OUT_OF_RANGE(env, "pass is too large");
+  //   return v8::Nothing<void>();
+  // }
+  // if (!salt.CheckSizeInt32()) [[unlikely]] {
+  //   THROW_ERR_OUT_OF_RANGE(env, "salt is too large");
+  //   return v8::Nothing<void>();
+  // }
+
+  auto paramsObj = args[offset + 1].As<v8::Object>();
+  auto propertyNames = paramsObj->GetOwnPropertyNames(env->context()).ToLocalChecked();
+
+  for (uint32_t i = 0; i < propertyNames->Length(); ++i) {
+    auto key = propertyNames->Get(env->context(), i).ToLocalChecked();
+    if (!key->IsString()) {
+      continue;
+    }
+
+    auto name = Utf8Value(env->isolate(), key).ToString();
+    auto value = paramsObj->Get(env->context(), key).ToLocalChecked();
+
+    if (value->IsUint32()) {
+      params->kdfParams[name] = value.As<v8::Uint32>()->Value();
+    } else if (value->IsNumber()) {
+      params->kdfParams[name] = value.As<v8::Number>()->Value();
+    } else if (value->IsString()) {
+      params->kdfParams[name] = Utf8Value(env->isolate(), value).ToString();
+    } else if (IsAnyBufferSource(value)) {
+      ArrayBufferOrViewContents<char> abv(value);
+      params->kdfParams[name] = std::vector<char>(abv.data(), abv.data() + abv.size());
+    }
+  }
+
+  if (!params->kdfCtx.setParams(params->kdfParams)) {
+    if (uint32_t err = ERR_peek_last_error(); err != 0) {
```

Comment:
This function should be using `ncrypto::ClearErrorOnReturn` or `ncrypto::MarkPopErrorOnReturn`